### PR TITLE
Add Class decoration

### PR DESCRIPTION
**Pull request by demix007**
### Interface

- [ ] Created a class Nameable.

- [ ] Implemented a method called correct_name that raises a NotImplementedError.
- [ ] Turned the Person class to Nameable.
- [ ] Made sure that the Person class inherits from Nameable

###  Base Decorator
- [ ] Made sure that it inherits from Nameable.
- [ ] Assigned a nameable object in the constructor from params to an instance variable.
- [ ] Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

### CapitalizeDecorator
- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implemented a method correct_name that capitalizes the output of @nameable.correct_name.

### TrimmerDecorator
- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

- [ ] Made sure that there were no Linter errors.
- [ ] Used the correct Gitflow.
- [ ] Documented my work in a professional way

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```
Comment (Deepakdanger) [inline trimmer_decorator.rb]: - [x] No need to initialize here, as it is inherited by BaseDecorator, so kindly remove it.

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  def correct_name
+    super[0..9]
+  end
```
Comment (Deepakdanger) [inline trimmer_decorator.rb]: - [x] kindly use the correct code to trim.
```
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @demix007 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (demix007) [reply trimmer_decorator.rb]: Thank you for pointing this change out, I have removed the initialize method just as you required.

File: trimmer_decorator.rb (hunk 2 above)
Comment (demix007) [reply trimmer_decorator.rb]: I have also implemented the change that you required here.

**Comment by demix007** [comment]
Hello @Deepakdanger 👋🏼 ,

Thank you for taking out time to review this and recommend changes to this PR. The recommended changes have been swiftly implemented. 👍🏼

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 🎉 

Hi, @demix007, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice class decoration 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by demix007** [comment]
## Hello @zieeco 👋🏼 , 

Thank you for taking out time to review and approve this  PR. You are amazing 💯

**Pull request merged**